service/store/store.go

```diff
@@ -517,30 +520,20 @@ func (s *Store) dealDataFilePathFor(bidID auction.BidID, payloadCid string) stri
 }
 
 // enqueueDataURI queues a data uri fetch.
-// commitTxn will be committed internally!
-func (s *Store) enqueueDataURI(commitTxn ds.Txn, b *Bid) error {
+func (s *Store) enqueueDataURI(txn ds.Txn, b *Bid) error {
 	// Set the bid to "fetching_data"
-	if err := s.saveAndTransitionStatus(commitTxn, b, BidStatusFetchingData); err != nil {
+	if err := s.saveAndTransitionStatus(txn, b, BidStatusFetchingData); err != nil {
 		return fmt.Errorf("updating status (fetching_data): %v", err)
 	}
-	if commitTxn != nil {
-		if err := commitTxn.Commit(); err != nil {
-			return fmt.Errorf("committing txn: %v", err)
+	select {
+	case s.jobCh <- b:
+	default:
+		log.Debugf("workers are busy; queueing %s", b.ID)
+		// Workers are busy, set back to "queued_data"
+		if err := s.saveAndTransitionStatus(nil, b, BidStatusQueuedData); err != nil {
```

Comment:
This will create a new txn before the other one (`txn` in this context) is committed... not sure how that will pan out, but I'd guess the first one gets priority?

Reply:
Oh wow so the status changes to `BidStatusQueuedData`, then to `BidStatusFetchingData` again when the `txn` in this context gets committed? Doesn't sound right... I can see why it commits internally now. I guess making this in the same transaction would work?